tests/src/psa_crypto_helpers.c

```diff
@@ -0,0 +1,46 @@
+/** \file psa_crypto_helpers.c
+ *
+ * \brief Helper functions to test PSA crypto functionality.
+ */
+
+/*
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <test/helpers.h>
+#include <test/macros.h>
+
+#if defined(MBEDTLS_PSA_CRYPTO_C)
+
+#include <psa/crypto.h>
+
+#if defined(MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG)
+#include <test/random.h>
+
+psa_status_t mbedtls_psa_external_get_random(
+    mbedtls_psa_external_random_context_t *context,
+    uint8_t *output, size_t output_size, size_t *output_length )
+{
+    /* This implementation is for test purposes only!
+     * Use the libc non-cryptographic random generator. */
```

Comment:
Using a non-crypto RNG (with a constant seed, too) is fine in unit tests. But I'm a bit concerned that if you enable `MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG` but don't supply a driver and build the whole Mbed TLS source, you also get sample programs, many of which could serve a useful purpose in a production environment. Should it be harder to include this function in the build? Surround it by `#if defined(TEST_PSA_CRYPTO_EXTERNAL_RNG)` (not listed in `config.h`)?

Reply:
I'm going to make this function return `PSA_ERROR_INSUFFICIENT_ENTROPY` by default. It's not perfect because it assumes applications check the return value of `psa_generate_random`, but it'll work in most cases since our library (including `psa_generate_key`) and example programs do fail if random generation fail.

Doing this is hitting some technical debt in the test helpers so I'll do it in a follow-up (either with or before #3833).

Reply:
Done in https://github.com/ARMmbed/mbedtls/pull/3912.